### PR TITLE
React-Group-Project

### PR DESCRIPTION
**Pull request by Fondem-Jr**
# In this development branch, we followed the project requirements, so the following implementations were adopted:

- [ ] Config & basic setup of React App;
- [ ] Created routes and view components (Rockets, Missions, Dangons and My Profile); we used for the page navigation links and styled active class to indicate which page user is currently on (underline active navigation link and a different shade of colour);
- [ ] Created directories for all Redux state slice files (rockets, missions, dragons);
- [ ] Redux: Fetch data and update Redux store;
- [ ] Upon the first render, we fetch data from the SpaceX API endpoints:
- Rockets: https://api.spacexdata.com/v3/rockets
- Missions: https://api.spacexdata.com/v3/missions
- Dragons: https://api.spacexdata.com/v3/dragons

- [ ] Once the data are fetched, we dispatch an action to store the selected data in Redux store;
NOTE: We use Rockets as the default view, so we fetch rockets data when the application starts;
- [ ] When a user clicks the "Reserve rocket" button, the action is dispatched to update the store.
- [ ] When a user clicks the "Reserve dragon" button, the action is dispatched to update the store.
- [ ] We created a reducer and action dispatcher for the "Join Mission" button;
- [ ] We wrote actions and reducers for cancelling rockets, dragons and leaving missions;
- [ ] Render UI: My Profile section;
- [ ] Compose two layouts and list only the rockets reserved, dragons reserved and missions joined by the use;
- [ ] Technical set up**:
- Set up the repository on GitHub and use Gitflow;
- Set up React and Redux;
- Set up testing libraries;
- We do not preserve user data;
For this project, we used Kanban board with GitHub project;

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,10 +1,37 @@
...
-    Hello React
-  </div>
-);
+const App = () => {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(getMissions());
+    dispatch(getDragons());
+    dispatch(getRockets());
+  }, [dispatch]);
```
Comment (omar-labana) [inline src/App.js:20]: - [ ] For each module, the view is currently showing only the loading status or the fetched results but what if the server didn't respond with data? For this case, I'd suggest that you guys conditionally render the `Rockets`, `Missions`, and `Dragons` to show a disclaimer if the server failed to respond or didn't send data at all.

**Review by omar-labana — CHANGES_REQUESTED** [review]
Hi team, 

Excellent work so far! 👌
You have correctly implemented all of the requirements and have demonstrated your understanding of React. :clap:. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️
- [ ] When visiting the `Missions` view for the first time, this error arises. Please fix it.
image

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js (hunk 1 above)
Comment (codepantherr) [reply src/App.js:20]: Hi @omar-labana! Thank you for pointing this out. We have implemented this change.

**Review by Lameck1 — APPROVED** [review]
Hi, @Fondem-Jr, @codepantherr & @JuliCarracedo,

You've done well working on the concerns of the previous review 👍🏼.
Your project is now complete and you are free to merge this branch :shipit: 

Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**